src/Umbraco.Core/Models/ContentRepositoryExtensions.cs

```diff
@@ -103,11 +102,11 @@ public static void CopyFrom(this IContent content, IContent other, string cultur
 
         public static void SetPublishInfo(this IContent content, string culture, string name, DateTime date)
         {
-            if (string.IsNullOrWhiteSpace(name))
-                throw new ArgumentNullOrEmptyException(nameof(name));
+            if (name == null) throw new ArgumentNullException(nameof(name));
```

Comment:
These null and then null/whitespace checks I think are unecessary. Both resharper and VS when using the tools will just do a single check and throw an `ArgumentException` which i think is fine. It's ok to leave these in here now that they are here but since the vast majority of us will be using VS or resharper tools most times checking for null/empty parameters will just be throwing a single ArgumentException.

Reply:
Only ReSharper does that: VS correctly suggests both `ArgumentNullException` and `ArgumentException` 😉

Reply:
As per comment here: https://github.com/umbraco/Umbraco-CMS/pull/5950#discussion_r317768334